node/core/pvf/src/execute/worker.rs

```diff
@@ -45,13 +47,21 @@ use std::{
 };
 
 /// Spawns a new worker with the given program path that acts as the worker and the spawn timeout.
+/// Sends a handshake message to the worker as soon as it is spawned.
 ///
 /// The program should be able to handle `<program-path> execute-worker <socket-path>` invocation.
 pub async fn spawn(
 	program_path: &Path,
+	executor_params: ExecutorParams,
 	spawn_timeout: Duration,
 ) -> Result<(IdleWorker, WorkerHandle), SpawnErr> {
-	spawn_with_program_path("execute", program_path, &["execute-worker"], spawn_timeout).await
+	let (mut idle_worker, worker_handle) =
+		spawn_with_program_path("execute", program_path, &["execute-worker"], spawn_timeout)
+			.await?;
+	send_handshake(&mut idle_worker.stream, Handshake { executor_params })
+		.await
+		.map_err(|_| SpawnErr::Handshake)?;
```

Comment:
nit: Worth logging the original error?

Reply:
I'm still struggling to understand how `gum` works, and I doubt if logging here adds to observability (as we're lacking validation code hash and candidate hash at this point), but I'll add it anyway. More logging is better than less logging :slightly_smiling_face:

Reply:
Thanks! I don't like discarding errors in general, as we lose information; might as well log it. Actually, I'm always careful with `_`, especially as a reviewer, because I have to slow down and make sure we're not tossing away something important. Might just be me being too nitty though.